tap_hubspot/client.py

```diff
@@ -115,14 +129,72 @@ def get_url_params(
             A dictionary of URL query parameters.
         """
         params: dict = {}
+        params["limit"] = 100
         if next_page_token:
             params["after"] = next_page_token
         if self.replication_key:
             params["sort"] = "asc"
             params["order_by"] = self.replication_key
-
         return params
 
+    def _is_incremental_search(self, context):
+        return self.replication_method == REPLICATION_INCREMENTAL and self.get_starting_replication_key_value(context) and hasattr(self, "incremental_path") and self.incremental_path
+
+    def prepare_request_payload(
+        self,
+        context: dict | None,
+        next_page_token: _TToken | None,
+    ) -> dict | None:
+        """Prepare the data payload for the REST API request.
+
+        By default, no payload will be sent (return None).
+
+        Developers may override this method if the API requires a custom payload along
+        with the request. (This is generally not required for APIs which use the
+        HTTP 'GET' method.)
+
+        Args:
+            context: Stream partition or context dictionary.
+            next_page_token: Token, page number or any request argument to request the
+                next page of data.
+        """
+        body = {}
+        if self._is_incremental_search(context):
+            # Only filter in case we have a value to filter on
+            # https://developers.hubspot.com/docs/api/crm/search
+            ts = datetime.datetime.strptime(self.get_starting_replication_key_value(context), "%Y-%m-%dT%H:%M:%S.%fZ")
+            # The SDK rounds up when handling datetimes sometimes so we need to subtract a second to be safe
+            ts = ts - datetime.timedelta(seconds=1)
```

Comment:
I'm curious what the raw value returned by the API is, what the SDK serializes it to, and what the SDK deserializes it to when reading from state.

Reply:
Its been a couple weeks since I was working on this and cant remember what I was doing to see this but I definitely remember seeing it. It could have been a bug on my size or a configuration issue. For now I removed it and will create an issue if I see it again.